module/Activity/src/Activity/Form/Activity.php

```diff
@@ -0,0 +1,202 @@
+            'attributes' => [
+                'type' => 'text',
+                'style' => 'width:100%',
+            ],
+        ]);
+
+        $this->add([
+            'type' => 'Zend\Form\Element\DateTime',
+            'name' => 'beginTime',
+            'attributes' => [
+                'min' => '2010-01-01T00:00:00Z',
+                'step' => '1', // minutes; default step interval is 1 min
+                'style' => 'width:100%',
+            ],
+        ]);
+
+        $this->add([
+            'type' => 'Zend\Form\Element\DateTime',
+            'name' => 'endTime',
+            'attributes' => [
+                'min' => '2010-01-01T00:00:00Z',
+                'step' => '1', // minutes; default step interval is 1 min
+                'style' => 'width:100%',
+            ],
+        ]);
+
+        $this->add([
+            'name' => 'location',
+            'attributes' => [
+                'type' => 'text',
+                'style' => 'width:100%',
+            ],
+        ]);
+
+        $this->add([
+            'name' => 'costs',
+            'attributes' => [
+                'type' => 'text',
+                'style' => 'width:100%',
+            ],
+        ]);
+
+        $this->add([
+            'name' => 'approved',
+            'type' => 'Zend\Form\Element\Checkbox',
+            'options' => array(
+                'use_hidden_element' => true,
+                'checked_value' => 1,
+                'unchecked_value' => 0,
+            ),
+        ]);
+        $this->add([
+            'name' => 'description',
+            'attributes' => [
+                'type' => 'textarea',
+                'style' => 'width:100%; height:10em; resize:none',
+            ],
+        ]);
+        $this->add(array(
+            'name' => 'submit',
+            'attributes' => array(
+                'type' => 'submit',
+                'value' => 'Create',
+            ),
+        ));
+    }
+
+    /*************** INPUT FILTER*****************/
+    /** The code below this deals with the input filter
+     * of the create and edit activity form data.
+     */
+
+    /**
+     * Get the input filter.
+     *
+     * @return InputFilterInterface
+     */
+    public function getInputFilter()
+    {
+        // Check if the input filter is set. If so, serve
+        if ($this->inputFilter) {
+            return $this->inputFilter;
+        }
+
+        $inputFilter = new InputFilter();
+        $factory = new InputFactory();
+
+        $inputFilter->add($factory->createInput([
+            'name' => 'beginTime',
+            'required' => true,
+        ]));
+
+        $inputFilter->add($factory->createInput([
+            'name' => 'endTime',
+            'required' => true,
+        ]));
+
+        $inputFilter->add($factory->createInput([
+            'name' => 'name',
+            'required' => true,
+            'filters' => [
+                ['name' => 'StripTags'],
+                ['name' => 'StringTrim'],
+            ],
```

Comment:
Daar ben ik het helemaal mee eens. `Zend\Escaper` is er voor het context-aware strippen van data. Tags hoef je er vantevoren niet uit te halen dus.